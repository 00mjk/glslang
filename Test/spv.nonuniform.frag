#version 450

layout(location=0) nonuniformEXT in vec4 nu_inv4;
nonuniformEXT float nu_gf;

nonuniformEXT int foo(nonuniformEXT int nupi, nonuniformEXT out int f)
{
    return nupi;
}

void main()
{
    nonuniformEXT int nu_li;

    int a = foo(nu_li, nu_li);
    nu_li = nonuniformEXT(a) + nonuniformEXT(a * 2);

    float b;
    b = nu_inv4.x * nu_gf;
}