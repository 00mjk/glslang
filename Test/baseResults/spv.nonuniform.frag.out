spv.nonuniform.frag
WARNING: 0:8: '[]' : assuming array size of one for compile-time checking of binding numbers for unsized array 

// Module Version 10000
// Generated by (magic number): 80006
// Id's are bound by 57

                              Capability Shader
                              Capability CapabilityShaderNonUniformEXT
                              Capability CapabilityRuntimeDescriptorArrayEXT
                              Extension  "SPV_EXT_descriptor_indexing"
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 4  "main" 33 48
                              ExecutionMode 4 OriginUpperLeft
                              Source GLSL 450
                              SourceExtension  "GL_EXT_nonuniform_qualifier"
                              Name 4  "main"
                              Name 11  "foo(i1;i1;"
                              Name 9  "nupi"
                              Name 10  "f"
                              Name 16  "a"
                              Name 17  "nu_li"
                              Name 18  "param"
                              Name 20  "param"
                              Name 30  "b"
                              Name 33  "nu_inv4"
                              Name 39  "nu_gf"
                              Name 46  "s2d"
                              Name 48  "nu_ii"
                              Decorate 13 DecorationNonUniformEXT
                              Decorate 17(nu_li) DecorationNonUniformEXT
                              Decorate 19 DecorationNonUniformEXT
                              Decorate 23 DecorationNonUniformEXT
                              Decorate 26 DecorationNonUniformEXT
                              Decorate 33(nu_inv4) Location 0
                              Decorate 33(nu_inv4) DecorationNonUniformEXT
                              Decorate 39(nu_gf) DecorationNonUniformEXT
                              Decorate 40 DecorationNonUniformEXT
                              Decorate 46(s2d) DescriptorSet 0
                              Decorate 46(s2d) Binding 0
                              Decorate 48(nu_ii) Flat
                              Decorate 48(nu_ii) Location 1
                              Decorate 48(nu_ii) DecorationNonUniformEXT
                              Decorate 49 DecorationNonUniformEXT
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeInt 32 1
               7:             TypePointer Function 6(int)
               8:             TypeFunction 6(int) 7(ptr) 7(ptr)
              25:      6(int) Constant 2
              28:             TypeFloat 32
              29:             TypePointer Function 28(float)
              31:             TypeVector 28(float) 4
              32:             TypePointer Input 31(fvec4)
     33(nu_inv4):     32(ptr) Variable Input
              34:             TypeInt 32 0
              35:     34(int) Constant 0
              36:             TypePointer Input 28(float)
              42:             TypeImage 28(float) 2D sampled format:Unknown
              43:             TypeSampledImage 42
              44:             TypeRuntimeArray 43
              45:             TypePointer UniformConstant 44
         46(s2d):     45(ptr) Variable UniformConstant
              47:             TypePointer Input 6(int)
       48(nu_ii):     47(ptr) Variable Input
              50:             TypePointer UniformConstant 43
              53:             TypeVector 28(float) 2
              54:   28(float) Constant 1056964608
              55:   53(fvec2) ConstantComposite 54 54
         4(main):           2 Function None 3
               5:             Label
           16(a):      7(ptr) Variable Function
       17(nu_li):      7(ptr) Variable Function
       18(param):      7(ptr) Variable Function
       20(param):      7(ptr) Variable Function
           30(b):     29(ptr) Variable Function
       39(nu_gf):     29(ptr) Variable Function
              19:      6(int) Load 17(nu_li)
                              Store 18(param) 19
              21:      6(int) FunctionCall 11(foo(i1;i1;) 18(param) 20(param)
              22:      6(int) Load 20(param)
                              Store 17(nu_li) 22
                              Store 16(a) 21
              23:      6(int) Load 16(a)
              24:      6(int) Load 16(a)
              26:      6(int) IMul 24 25
              27:      6(int) IAdd 23 26
                              Store 17(nu_li) 27
              37:     36(ptr) AccessChain 33(nu_inv4) 35
              38:   28(float) Load 37
              40:   28(float) Load 39(nu_gf)
              41:   28(float) FMul 38 40
                              Store 30(b) 41
              49:      6(int) Load 48(nu_ii)
              51:     50(ptr) AccessChain 46(s2d) 49
              52:          43 Load 51
              56:   31(fvec4) ImageSampleImplicitLod 52 55
                              Return
                              FunctionEnd
  11(foo(i1;i1;):      6(int) Function None 8
         9(nupi):      7(ptr) FunctionParameter
           10(f):      7(ptr) FunctionParameter
              12:             Label
              13:      6(int) Load 9(nupi)
                              ReturnValue 13
                              FunctionEnd
